src/test/java/com/exasol/adapter/document/files/AbsTestContainerIT.java

```diff
@@ -0,0 +1,65 @@
+package com.exasol.adapter.document.files;
+
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.util.List;
+
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import com.azure.core.util.BinaryData;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.exasol.adapter.document.files.abstestsetup.AbsTestSetup;
+import com.exasol.adapter.document.files.abstestsetup.OnlineAbsTestSetup;
+
+@Tag("integration")
+class AbsTestContainerIT {
+
+    static AbsTestSetup absTestSetup = new OnlineAbsTestSetup();
+
+    @Test
+    void emptyFilledContainer() {
+        try (final AbsTestContainer testContainer = AbsTestContainer.create(absTestSetup)) {
+            final BlobContainerClient client = testContainer.getBlobContainerClient();
+            client.getBlobClient("blobName").upload(BinaryData.fromString("blob content"));
+            assertThat(listBlobNames(client), contains("blobName"));
+            testContainer.empty();
+            assertThat(listBlobNames(client), empty());
+        }
+    }
+
+    // Note: This test will fail when Azure Blob Storage Gen2 is enabled.
+    @Test
+    void emptyFilledContainerWithDirectory() {
+        try (final AbsTestContainer testContainer = AbsTestContainer.create(absTestSetup)) {
+            final BlobContainerClient client = testContainer.getBlobContainerClient();
+            client.getBlobClient("dir/blobName").upload(BinaryData.fromString("blob content"));
+            assertThat(listBlobNames(client), contains("dir/blobName"));
+            testContainer.empty();
+            assertThat(listBlobNames(client), empty());
+        }
+    }
+
+    @Test
```

Comment:
I think you should change this method from a test into an assumption run as early as possible. I assume you want the whole integration test to fail if the precondition is not fulfilled.

Reply:
A direct check for the Gen2 would be preferred here of course.

Reply:
An assumption will skip the test, I want it to fail.
I didn't find a direct way to check if Gen2 is enabled.